deploy.py

```diff
@@ -20,22 +20,129 @@
 from __future__ import print_function
 
 import argparse
+import collections
 import copy
 import os
 import shutil
 import subprocess
 import sys
 
 import colorama
+from six.moves import input
 from tockloader import tab
 from tockloader import tbfh
 from tockloader import tockloader as loader
 from tockloader.exceptions import TockLoaderException
 
-# This structure allows us in the future to also support out-of-tree boards.
+PROGRAMMERS = frozenset(("jlink", "openocd", "pyocd", "nordicdfu", "none"))
+
+# This structure allows us to support out-of-tree boards as well as (in the
+# future) more achitectures.
+OpenSKBoard = collections.namedtuple(
+    "OpenSKBoard",
+    [
+        # Location of the Tock board (where the Makefile file is
+        "path",
+        # Targeted architecture (e.g. thumbv7em-none-eabi)
```

Comment:
I think "target" is a more common spelling in this context.

Reply:
Done